### PR TITLE
[ms-gsl] Update to v3.0.0

### PR DESCRIPTION
**Pull request by Farwaykorse**
[Microsoft GSL](https://github.com/microsoft/gsl) update to release v3.0.0.

[Release notes](https://github.com/microsoft/GSL/releases/tag/v3.0.0)

----
### Changes to the portfile
Add support for CMake `find_package()` (besides `find_path()`):
```CMake
find_package(Microsoft.GSL CONFIG REQUIRED)

target_link_libraries(main PRIVATE Microsoft.GSL::GSL)
```

### CODE REVIEW

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

File: ports/ms-gsl/portfile.cmake [hunk 1]
```diff
@@ -1,13 +1,30 @@
...
+vcpkg_configure_cmake(
+    SOURCE_PATH ${SOURCE_PATH}
+    PREFER_NINJA
+    OPTIONS
+        -DGSL_TEST=OFF
+)
+
+vcpkg_install_cmake()
+
 file(INSTALL ${SOURCE_PATH}/include/ DESTINATION ${CURRENT_PACKAGES_DIR}/include)
```
[the remark this answers is not in the document]
Comment (Farwaykorse) [reply ports/ms-gsl/portfile.cmake:19]: Sharp!

File: ports/ms-gsl/portfile.cmake [hunk 2]
```diff
@@ -1,13 +1,30 @@
...
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug")
 
 # Handle copyright
 file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/ms-gsl RENAME copyright)
```
[the remark this answers is not in the document]
Comment (Farwaykorse) [reply ports/ms-gsl/portfile.cmake]: Nice!

**Review by NancyLi1013 — APPROVED** [review]

File: ports/ms-gsl/portfile.cmake [hunk 3]
```diff
@@ -1,13 +1,29 @@
...
 # Handle copyright
-file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/ms-gsl RENAME copyright)
+file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
+
+# Move package config file (temporary fix)
+file(GLOB GSLFILEFOUND "${CURRENT_PACKAGES_DIR}/share/Microsoft.GSL/cmake/*.cmake")
+if(GSLFILEFOUND)
+    file(INSTALL ${GSLFILEFOUND} DESTINATION "${CURRENT_PACKAGES_DIR}/share/Microsoft.GSL")
+    file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/share/Microsoft.GSL/cmake")
+endif()
```
Comment (Neumann-A) [inline ports/ms-gsl/portfile.cmake:29]: `vcpkg_fixup_cmake_targets()` instead and above `file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug")`

Comment (Farwaykorse) [reply ports/ms-gsl/portfile.cmake:29]: Done, but it is kinda strange since `share/Microsoft.GSL/cmake` is actually a valid path format for CMake. #10881

Comment (Neumann-A) [reply ports/ms-gsl/portfile.cmake]: I think its just a bug in the usage display. The `find_package(Microsoft.GSL CONFIG REQUIRED)` should work nevertheless or there is a bug in CMakes documentation

**Comment by ras0219-msft** [comment]
Thanks for the PR!

**Pull request merged**